src/observers/link_prefetch_observer.js

```diff
@@ -145,50 +139,52 @@ export class LinkPrefetchObserver {
   #isPrefetchable(link) {
     const href = link.getAttribute("href")
 
-    if (!href || href.startsWith("#") || link.getAttribute("data-turbo") === "false" || link.getAttribute("data-turbo-prefetch") === "false") {
-      return false
-    }
+    if (!href) return false
 
-    const event = dispatch("turbo:before-prefetch", {
-      target: link,
-      cancelable: true
-    })
+    if (unfetchableLink(link)) return false
+    if (linkToTheSamePage(link)) return false
+    if (linkOptsOut(link)) return false
+    if (nonSafeLink(link)) return false
+    if (eventPrevented(link)) return false
```

Comment:
This is much nicer with these checks extracted and given meaningful names, btw! 👏